numpy/core/arrayprint.py

```diff
@@ -521,7 +515,9 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
-    if a.size == 0:
+    if style is not None and a.shape == ():
+        return style(a[()])
```

Comment:
I think we can kill this `if` entirely now, and leave the `style` argument as obsolete?